### PR TITLE
Minor VectorAPI boxing fixes

### PR DESCRIPTION
**Pull request by gita-omr**
- do not use TR_enableMaskUnboxing environment option since the issue was fixed
- handle astore correctly when RHS is not aload

### CODE REVIEW

**Comment by gita-omr** [comment]
Removed unneeded `()`

File: runtime/compiler/optimizer/VectorAPIExpansion.cpp [hunk 1]
```diff
@@ -1802,8 +1798,17 @@ TR_VectorAPIExpansion::transformIL(bool checkBoxing)
...
          numLanes = bitsLength/8/elementSize;
 
-         // TODO: add an assert that unboxing is not needed in this case since the temp
-         // would not be vectorized
+         if (boxingAllowed())
+            {
+            TR::DataType elementTypeTmp;
+            int32_t bitsLengthTmp;
+            vapiObjType objectTypeTmp;
+            bool scalarizedTmp;
```
Comment (zl-wang) [inline runtime/compiler/optimizer/VectorAPIExpansion.cpp:1806]: you don't need to initialize these 4 Tmp(s)? and, they are not used after coming back from the call.  we experienced a very elusive bug when a call is made with arguments not initialized (although the code was correct): OpenXL optimizer just generated garbage code.

Comment (gita-omr) [reply runtime/compiler/optimizer/VectorAPIExpansion.cpp:1806]: These are values set by the call and they are not used in this case. Not sure how to make that nicer.

Comment (gita-omr) [reply runtime/compiler/optimizer/VectorAPIExpansion.cpp:1806]: We do it in other places too. In the future we can return a structure filled in with those values.

Comment (gita-omr) [reply runtime/compiler/optimizer/VectorAPIExpansion.cpp:1806]: For now, I gave them *Tmp names to make it more obvious.

**Review by zl-wang — APPROVED** [review]

**Comment by zl-wang** [comment]
Jenkins test sanity.functional zlinux,plinux,xlinux jdk23

**Pull request merged**